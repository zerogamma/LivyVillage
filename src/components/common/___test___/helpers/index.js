import React from 'react'
import renderer from 'react-test-renderer'
import theme from '../../../../utils/style/theme'
import { ThemeProvider } from 'styled-components'

export function renderWithTheme(component) {
    return renderer.create(
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
    )
}
