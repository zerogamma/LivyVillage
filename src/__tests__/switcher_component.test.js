import React from "react"
import Switcher from "../components/village/switcher";


import 'jest-styled-components';
import { renderWithTheme } from '../components/common/___test___/helpers';

describe("Switcher Component render test", () => {
    it("renders Switcher component correctly", () => {
      const data = 5;
      const comp = renderWithTheme(<Switcher noData={false} page={data} setPage={()=>{}}/>).toJSON()
      expect(comp).toMatchSnapshot()
    })
  })