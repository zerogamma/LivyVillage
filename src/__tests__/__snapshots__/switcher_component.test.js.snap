// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher Component render test renders Switcher component correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 2em !important;
  height: 2em !important;
  color: black;
}

.c2 {
  width: 2em !important;
  height: 2em !important;
}

@media (max-width:767px) {
  .c0 {
    margin-top: 15px;
  }
}

@media (max-width:767px) {
  .c1 {
    width: 1em !important;
    height: 1em !important;
  }
}

@media (max-width:767px) {
  .c2 {
    width: 1em !important;
    height: 1em !important;
  }
}

<div
  className="c0"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root c1"
    disabled={false}
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M6 6h2v12H6zm3.5 6l8.5 6V6z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root c2"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M6 18l8.5-6L6 6v12zM16 6v12h2V6h-2z"
    />
  </svg>
</div>
`;
